src/lib.rs

```diff
@@ -824,73 +832,84 @@ impl<M, R, S> Client<M, R, S> {
     }
 
     /// Performs a request for which the server returns a oneshot receiver.
-    pub async fn rpc<Req, Res>(&self, msg: Req) -> Result<Res, Error>
+    pub fn rpc<Req, Res>(
+        &self,
+        msg: Req,
+    ) -> impl Future<Output = Result<Res, Error>> + Send + 'static
     where
         S: Service,
         M: From<WithChannels<Req, S>> + Send + Sync + Unpin + 'static,
```

Comment:
R could have the bound `RpcMessage`, and M could have the bound `LocalRpcMessage` to DRY the bounds. Not sure if this is worth it. I guess it depends a bit on if we need to take away some of the bounds for WASM. @matheus23 WDYT?

Reply:
Sorry, I have a really hard time reading this type signature.

In general, if you're wondering if it's okay to have a `T: Send` bound in Wasm, then you simply need to answer this question: "Will `T` contain/refer to a `JsValue`?"
Such a value could be a `JsPromise` if it just made a call to sth async (this happens a lot when you have async APIs and is what makes rust futures in async non-`Send` in practice a lot), or it could be because you're storing a js `Error` value in there, which is the other common thing.

I'd very hesitantly say you're good in this case to just have a `Send` bound. But again, I have no idea what's going on in this type signature, it's quite complex.

Reply:
Will T contain/refer to a JsValue?

So if there is a possibility that `T` contains a JsValue, it must not be `Send`? I think it is very unlikely that there will be a JsValue in these things.